cfgov/legacy/housing_counselor/generator.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import absolute_import, unicode_literals
+
+import glob
+import json
+import os
+import re
+import sqlite3
+
+from django.template import loader
+from math import acos, cos, radians, sin
+from six import print_
+
+
+def distance_in_miles(lat1_radians, lng1_radians, lat2_radians, lng2_radians):
+    """Estimate distance in miles between two points in radians.
+
+    Uses simplified algorithm to determine proximity based on latitude and
+    longitude, described at https://stackoverflow.com/q/1916953.
+    """
+    earth_radius_in_miles = 3959
+
+    return earth_radius_in_miles * acos(
+        cos(lat1_radians) *
+        cos(lat2_radians) *
+        cos(lng2_radians - lng1_radians) +
+        sin(lat1_radians) *
+        sin(lat2_radians)
+    )
+
+
+def get_db_connection():
+    connection = sqlite3.connect(':memory:')
+    connection.create_function('distance_in_miles', 4, distance_in_miles)
+
+    return connection
+
+
+def fill_db(connection, counselors):
+    def prepare_counselors(counselors):
```

Comment:
@rosskarchner thanks for the review; see the second item at GHE#989. This isn't needed right now because we currently do a geo-based lookup at the HUD API, which only returns counselors that have coordinates (presumably that's how they get retrieved). If we wanted to change to retrieve all counselors, even those without coordinates (which, again, we don't do currently), then, yes, we'd have to geocode those.